core-bundle/src/Filesystem/PublicUri/SymlinkedLocalFilesProvider.php

```diff
@@ -13,17 +13,32 @@
 namespace Contao\CoreBundle\Filesystem\PublicUri;
 
 use League\Flysystem\FilesystemAdapter;
+use League\Flysystem\Local\LocalFilesystemAdapter;
 use Nyholm\Psr7\Uri;
 use Psr\Http\Message\UriInterface;
 use Symfony\Component\HttpFoundation\RequestStack;
 
 class SymlinkedLocalFilesProvider implements PublicUriProviderInterface
 {
+    /**
+     * @var array<string, string>
+     */
+    private array $adapters = [];
+
     public function __construct(
-        private readonly FilesystemAdapter $localFilesAdapter,
-        private readonly string $uploadDir,
+        LocalFilesystemAdapter $localFilesAdapter,
+        readonly string $uploadDir,
         private readonly RequestStack $requestStack,
     ) {
+        $this->registerAdapter($localFilesAdapter, $uploadDir);
+    }
+
+    /**
+     * @internal
+     */
+    public function registerAdapter(LocalFilesystemAdapter $adapter, string $rootPath): void
+    {
+        $this->adapters[spl_object_hash($adapter)] = $rootPath;
```

Comment:
Is there a specific reason using object hashes here instead of storing it in a `WeakMap`?

Reply:
You'right, a weak map would nicely fit here. :slightly_smiling_face: 
Changed in ee49e7501907e08a3996dfe9d8b6632549650337.